mindmeld/components/entity_resolver.py

```diff
@@ -1693,6 +1368,63 @@ def _load(self):
         self.fit()
 
 
+class SentenceBertCosSimEntityResolver(EmbedderCosSimEntityResolver):
+    """
+    Resolver class for bert models as described here:
+    https://github.com/UKPLab/sentence-transformers
+    """
+
+    def __init__(self, app_path, **kwargs):
+        """
+        This wrapper class allows creation of a BERT base embedder class
+        (currently based on sentence-transformers)
+
+        Specificall, this wrapper updates er_config in kwargs with
+            - any default settings if unavailable in input
+            - cache path
+
+        Args:
+            app_path (str): App's path to cache embeddings
+            er_config (dict): Configurations can be passed in through `model_settings` field
+                `model_settings`:
+                    embedder_type: the type of embedder picked from embedder_models.py class
+                        (eg. 'bert', 'glove', etc. )
+                    pretrained_name_or_abspath: the pretrained model for 'bert' embedder
+                    bert_output_type: if the output is a sentence mean pool or CLS output
+                    quantize_model: if the model needs to be qunatized for faster inference time
+                        but at a possibly reduced accuracy
+                    concat_last_n_layers: if some of the last layers of a BERT model are to be
+                        concatenated for better accuracies
+                    normalize_token_embs: if the obtained sub-token level encodings are to be
+                        normalized
+        """
+
+        # default configs useful for reusing model's encodings through a cache path
+        defaults = {
+            "embedder_type": "bert",
+            "pretrained_name_or_abspath": "distilbert-base-nli-stsb-mean-tokens",
+            "bert_output_type": "mean",
+            "quantize_model": True,
+            "concat_last_n_layers": 4,
+            "normalize_token_embs": True,
+        }
+
+        # determine cache path and pass it in `model_settings`
+        er_config = copy.deepcopy(kwargs.get("er_config", {}))
+        model_settings = er_config.get("model_settings", {})
+        for key, value in model_settings.items():
+            if key in defaults:
+                defaults.update({key: value})
+        string = json.dumps(defaults, sort_keys=True)
+        hashid = f"{Hasher(algorithm='sha1').hash(string=string)}$synonym_{kwargs['entity_type']}"
```

Comment:
lets use sha256 since it provides a lesser attack surface for hash collisions

Reply:
Good point. Will do that change!